### ISSUE TITLE
"func host start" gives "Array dimensions exceeded supported range." error

### ISSUE THREAD

**Issue by jmo808** [Azure/azure-functions-core-tools#704]
Using newest release of azure-functions-core-tools.

**Comment by jmo808** [comment]
If I try and build a container out of the function, it gives me a 401 error on the function route. ie. http://localhost:8080/api/testHttp?name=stuff. I've tried the node.js and python worker runtimes with the same problem.

**Comment by ljtill** [comment]
Encountering this issue with the latest release (2.0.1-beta.38).

**Comment by benc-uk** [comment]
Same problem. On Node 8.12.0 and running under WSL, can't start the host

I also created a new function app folder (and ran `func init`) and got same problem

Right now the local tools are unusable for me :(

**Comment by benc-uk** [comment]
I rolled back to version **2.0.1-beta.35** and the problem went away, but beta37 and beta38 are a total bust

**Comment by ahmelsayed** [comment]
can you set `CLI_DEBUG=1` env var and run the same command again and share the stack trace?

**Comment by benc-uk** [comment]
I get this
```
System.OverflowException: Array dimensions exceeded supported range.
   at System.Net.NetworkInformation.StringParsingHelpers.ParseActiveTcpConnectionsFromFiles(String tcp4ConnectionsFile, String tcp6ConnectionsFile)
   at System.Net.NetworkInformation.LinuxIPGlobalProperties.GetActiveTcpConnections()
   at Azure.Functions.Cli.Helpers.NetworkHelpers.IsPortAvailable(Int32 port)
   at Azure.Functions.Cli.Actions.HostActions.StartHostAction.PreRunConditions(WorkerRuntime workerRuntime)
   at Azure.Functions.Cli.Actions.HostActions.StartHostAction.RunAsync()
   at Azure.Functions.Cli.ConsoleApp.RunAsync[T](String[] args, IContainer container)

```

**Comment by ljtill** [comment]
Installed the latest release (2.0.1-beta.39) and encountering the issue with that too.

**Comment by benc-uk** [comment]
Starting to think this is a WSL specific error. It works for me in PowerShell and even just tried inside a Azure Cloud Shell session and `func host start` worked there (obviously it's of no use but it was just a test!)

The error seems to be to do with enumerating network connections, which inside a WSL session are handled a little different from a regular Linux host

**Comment by benc-uk** [comment]
OK I've got to the bottom of this, it's caused by the checks being done in 
https://github.com/Azure/azure-functions-core-tools/blob/master/src/Azure.Functions.Cli/Helpers/NetworkHelpers.cs

And the root cause is actual a bug in .NET Core, when running under WSL
https://github.com/dotnet/corefx/issues/30909

The fix is pretty simple, with some error/exception handling wrapping the code **NetworkHelpers.cs** I'll try to get a PR together for this

**Comment by ahmelsayed** [comment]
Fixed by @benc-uk in #733 

Thanks Ben!
